File: LeetCode/14. Longest Common Prefix/solution.ts
// NOT CORRECT
function longestCommonPrefix(strs: string[]): string {
    let s = '';
    let i = 0;
    while (i < strs[0].length) {
        let allEqual = true;
        let currentChar = strs[0][i];
        for (let j = 1; j < strs.length; j++) {
            console.log('strs[j]', strs[j]);
            if (strs[j] !== currentChar) {
                allEqual = false;
                break;
            }
        }
        if (allEqual) s += currentChar;
        else {
            console.log(s);
            console.log(currentChar);
            
            
        }
        i++;
    }
    return s;
}


const strs1 = ["flower","flow","flight"];
console.log(longestCommonPrefix(strs1));

const strs2 = ["dog","racecar","car"];
console.log(longestCommonPrefix(strs2));

File: LeetCode/14. Longest Common Prefix/solution.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        i = 0
        common_prefix = ''
        while i < len(strs[0]):
            char_to_compare = strs[0][i]
            for word in strs:
                if i >= len(word) or word[i] != char_to_compare:
                    return common_prefix
            common_prefix += char_to_compare
            i += 1
        return common_prefix


    strs1 = ["flower","flow","flight"]
    strs2 = ["dog","racecar","car"]
    
    
    def test1(self):
        print(self.longestCommonPrefix(self.strs1))

    @staticmethod
    def hi():
        print('hi')
            
# --TESTS--

strs1 = ["flower","flow","flight"]
strs2 = ["dog","racecar","car"]
        
sol = Solution()
print(sol.longestCommonPrefix(strs1))
print(sol.longestCommonPrefix(strs2))

sol.test1()
